core-rust/state-manager/src/protocol/protocol_update.rs

```diff
@@ -0,0 +1,489 @@
+// This file contains the protocol update logic for specific protocol versions
+
+use node_common::locks::{LockFactory, RwLock, StateLock};
+use radix_engine::track::StateUpdates;
+use radix_engine::transaction::CostingParameters;
+use radix_engine_common::network::NetworkDefinition;
+use radix_engine_common::prelude::Decimal;
+use std::ops::Deref;
+use std::sync::Arc;
+
+use transaction::prelude::TransactionPayloadPreparable;
+use transaction::validation::{NotarizedTransactionValidator, ValidationConfig};
+use utils::btreemap;
+
+use crate::epoch_handling::EpochAwareAccuTreeFactory;
+use crate::mainnet_updates::MainnetProtocolUpdaterFactory;
+use crate::traits::{
+    CommitBundle, CommitStore, CommittedTransactionBundle, HashTreeUpdate, QueryableProofStore,
+    ReceiptAccuTreeSliceV1, SubstateStoreUpdate, TransactionAccuTreeSliceV1,
+};
+use crate::transaction::{
+    ExecutionConfigurator, FlashTransactionV1, LedgerTransaction, LedgerTransactionValidator,
+    PreparedLedgerTransaction, TransactionSeriesExecutor,
+};
+use crate::{
+    CommittedTransactionIdentifiers, ExecutionCache, LedgerHeader, LedgerProof, LedgerProofOrigin,
+    LoggingConfig, ProtocolState, StateManagerDatabase,
+};
+
+pub trait ProtocolUpdaterFactory {
+    fn supports_protocol_version(&self, protocol_version_name: &str) -> bool;
```

Comment:
sounds good, they are lightweight

Reply:
nah, I think it's actually uglier without this extra method because `updater_for` requires to provide a store (which isn't necessary if we just want to sanity check the updater factory)

Reply:
@LukasGasior1 but [some other comments](https://github.com/radixdlt/babylon-node/pull/790#discussion_r1453378751) (from me and @talekhinezh) already point out that `updater_for()` maybe should not really take `store` as param.